src/balancecommand.cpp

```diff
@@ -0,0 +1,188 @@
+/****************************************************************************************
+**
+** Copyright (C) 2015 Jolla Ltd.
+** Contact: Martin Grimme <martin.grimme@gmail.com>
+** All rights reserved.
+**
+** This file is part of the btrfs-balancer package.
+**
+** You may use this file under the terms of the GNU Lesser General
+** Public License version 2.1 as published by the Free Software Foundation
+** and appearing in the file license.lgpl included in the packaging
+** of this file.
+**
+** This library is free software; you can redistribute it and/or
+** modify it under the terms of the GNU Lesser General Public
+** License version 2.1 as published by the Free Software Foundation
+** and appearing in the file license.lgpl included in the packaging
+** of this file.
+**
+** This library is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+** Lesser General Public License for more details.
+**
+****************************************************************************************/
+
+
+#include "balancecommand.h"
+#include "btrfsbalancer.h"
+#include "dbusservice.h"
+
+#include <QDBusConnection>
+#include <QDBusInterface>
+#include <QDBusMessage>
+#include <QFile>
+
+#include <iostream>
+
+namespace
+{
+const QString BATTERY_CHARGE("/run/state/namespaces/Battery/ChargePercentage");
```

Comment:
Or maybe instead of hardcoding this, use [statefs' contextsubscriber](https://github.com/nemomobile/statefs-contextkit/blob/master/include/contextsubscriber/contextproperty.h)?